internal/message/message.go

```diff
@@ -121,7 +123,16 @@ func (m *message) bind(parent gtk.Widgetter) *gio.Menu {
 		actions["message.delete"] = func() { m.view().Delete(m.message.ID) }
 	}
 
+	if channel != nil && (channel.Type == discord.DirectMessage || channel.Type == discord.GroupDM) {
+		actions["message.add-reaction"] = func() { m.ShowEmojiChooser() }
+	}
+
+	if state.HasPermissions(m.message.ChannelID, discord.PermissionAddReactions) {
```

Comment:
This might do network calls. You should use `state.Offline().HasPermissions()` (iirc).